lib/Common/Memory/CollectionState.h

```diff
@@ -39,6 +39,12 @@ enum CollectionState
     Collection_PostSweepRedeferralCallback = 0x00040000,
     Collection_WrapperCallback             = 0x00080000,
 
+#if ENABLE_CONCURRENT_GC && ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP
+    Collection_ConcurrentSweepPass1 = 0x00100000,
```

Comment:
Can we have more descriptive names other than Pass1 and Pass2? What actually happens to a heap block during Pass1 sweep and what happens during Pass2 sweep? At the very least, we should have that documented here if we can't come up with better names

Reply:
I will keep the names and add some documentation here as well. I have added details about the 2-pass sweep in the documentation for the method PrepareForAllocationsDuringConcurrentSweep.

---
In reply to: [148163410](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148163410) [](ancestors = 148163410)